theseus/src/api/profile_create.rs

```diff
@@ -0,0 +1,126 @@
+//! Theseus profile management interface
+use crate::{prelude::ModLoader, profile};
+pub use crate::{
+    state::{JavaSettings, Profile},
+    State,
+};
+use daedalus::modded::LoaderVersion;
+use dunce::canonicalize;
+use futures::prelude::*;
+use std::path::PathBuf;
+use tokio::fs;
+use tokio_stream::wrappers::ReadDirStream;
+
+const DEFAULT_NAME: &'static str = "Untitled Instance";
+
+// Generic basic profile creation tool.
+// Creates an essentially empty dummy profile with profile_create
+#[tracing::instrument]
+pub async fn profile_create_empty() -> crate::Result<Profile> {
+    Ok(profile_create(
+        String::from(DEFAULT_NAME), // the name/path of the profile
+        String::from("1.8.2"),      // the game version of the profile
+        ModLoader::Vanilla,         // the modloader to use
+        String::from("stable"), // the modloader version to use, set to "latest", "stable", or the ID of your chosen loader
+        None,                   // the icon for the profile
+    )
+    .await?)
+}
+
+// Creates a profile at  the given filepath and adds it to the in-memory state
+// This is reused mostly from the CLI. TODO: touch up.
+// invoke('profile_add',profile)
+#[tracing::instrument]
+pub async fn profile_create(
+    name: String,           // the name of the profile, and relative path
+    game_version: String,   // the game version of the profile
+    modloader: ModLoader,   // the modloader to use
+    loader_version: String, // the modloader version to use, set to "latest", "stable", or the ID of your chosen loader
+    icon: Option<PathBuf>,  // the icon for the profile
+) -> crate::Result<Profile> {
+    let state = State::get().await?;
+
+    let path = state.directories.profiles_dir().join(&name);
```

Comment:
For the profile directory I think using the name for the directory might be a bad idea as it limits what the user can name their instances (iirc FS stuff has to be UTF-8). Could we add a UUID field to profiles and use that as the directory name?